main.cpp

```diff
@@ -1301,7 +1307,7 @@ void push_message(int type, uint32_t lval, float fval, const char* sval) {
 
 			if (os.mqtt.enabled()) {
 				sprintf_P(topic, PSTR("opensprinkler/station/%d"), lval);
-				strcpy_P(payload, PSTR("{\"state\":1}"));
+				sprintf_P(payload, PSTR("{\"state %d\":1}"), lval);
```

Comment:
It's a good note that the full topic is available (I suppose that's the result of -v?) I agree that the station number should be included as a separate variable, like 'id', instead of being included in the state variable.

Reply:
I agree with @PeteBa that it's useless to repeat in the payload an information that is already in the topic

Reply:
Anyhow, in general, an MQTT payload should not contain identifiers, this should always go in the topic